Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -3067,7 +3063,9 @@ namespace OpenInfraPlatform
 					// Interpret parameter
 					double sineTerm = sine->SineTerm * this->UnitConvert()->getLengthInMeterFactor();
 					//if length of the curve is less than 1, factor is equal to 1, otherways factor is equal to lenght 
-					double factor = length;
+					double factor;
+					if (length < 1)factor = 1;
+					else factor = length;

```

Comment:
Apply here as well.

Reply:
Resolved in [d580137](https://github.com/tumcms/Open-Infra-Platform/pull/553/commits/d580137f72d12e4dea27ab4c1f0936a7c180c8b0)